packages/flutter_tools/lib/src/test/web_test_compiler.dart

```diff
@@ -47,24 +44,23 @@ class WebTestCompiler {
   final Config _config;
 
   Future<WebMemoryFS> initialize({
-    @required Directory projectDirectory,
-    @required String testOutputDir,
-    @required List<String> testFiles,
-    @required BuildInfo buildInfo,
+    required Directory projectDirectory,
+    required String testOutputDir,
+    required List<String> testFiles,
+    required BuildInfo buildInfo,
   }) async {
     LanguageVersion languageVersion = LanguageVersion(2, 8);
-    HostArtifact platformDillArtifact;
+    HostArtifact platformDillArtifact = HostArtifact.webPlatformSoundKernelDill;
```

Comment:
Default initialization to `webPlatformSoundKernelDill`, pulled up from `NullSafetyMode.sound` `if` statement.